sentry-android-replay/src/test/java/io/sentry/android/replay/capture/SessionCaptureStrategyTest.kt

```diff
@@ -367,4 +371,39 @@ class SessionCaptureStrategyTest {
                 "the current replay cache folder is not being deleted."
         )
     }
+
+    @Test
+    fun `records replay options event for segment 0`() {
+        fixture.options.experimental.sessionReplay.sessionSampleRate = 1.0
+        fixture.options.experimental.sessionReplay.maskAllImages = false
+        fixture.options.experimental.sessionReplay.quality = HIGH
+        fixture.options.experimental.sessionReplay.addMaskViewClass("my.custom.View")
+
+        val now =
+            System.currentTimeMillis() + (fixture.options.experimental.sessionReplay.sessionSegmentDuration * 5)
+        val strategy = fixture.getSut(dateProvider = { now })
+        strategy.start(fixture.recorderConfig)
+
+        strategy.onScreenshotRecorded(mock<Bitmap>()) {}
+
+        verify(fixture.hub).captureReplay(
+            argThat { event ->
+                event is SentryReplayEvent && event.segmentId == 0
+            },
+            check {
+                val optionsEvent =
+                    it.replayRecording?.payload?.filterIsInstance<RRWebOptionsEvent>()!!
+                assertEquals("sentry.java", optionsEvent[0].optionsPayload["nativeSdkName"])
+                assertEquals(BuildConfig.VERSION_NAME, optionsEvent[0].optionsPayload["nativeSdkVersion"])
+
+                assertEquals(null, optionsEvent[0].optionsPayload["errorSampleRate"])
+                assertEquals(1.0, optionsEvent[0].optionsPayload["sessionSampleRate"])
+                assertEquals(true, optionsEvent[0].optionsPayload["maskAllText"])
+                assertEquals(false, optionsEvent[0].optionsPayload["maskAllImages"])
+                assertEquals("high", optionsEvent[0].optionsPayload["quality"])
+                assertEquals("android.widget.TextView,android.webkit.WebView,android.widget.VideoView,androidx.media3.ui.PlayerView,com.google.android.exoplayer2.ui.PlayerView,com.google.android.exoplayer2.ui.StyledPlayerView,my.custom.View", optionsEvent[0].optionsPayload["maskedViewClasses"])
```

Comment:
AFAIU from your definition of optionsPayload value, we could add a list here instead of a concatenated string. Would it make sense to do this by default so we can potentially have a better UI for this?

Reply:
yeah just tried it and it looks fine! I will ask about the frontend change
![Google Chrome 2024-12-30 23 07 44](https://github.com/user-attachments/assets/2ea3a7f5-f614-445f-a833-029fcac92efb)